sentry-android-core/src/main/java/io/sentry/android/core/performance/TimeSpan.java

```diff
@@ -0,0 +1,164 @@
+package io.sentry.android.core.performance;
+
+import android.os.SystemClock;
+import io.sentry.DateUtils;
+import io.sentry.SentryDate;
+import io.sentry.SentryLongDate;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.annotations.TestOnly;
+
+/**
+ * A measurement for time critical components on a macro (ms) level. Based on {@link
+ * SystemClock#uptimeMillis()} to ensure linear time progression (as opposed to a syncable clock).
+ * To provide real world unix time information, the start uptime time is stored alongside the unix
+ * time. The stop unix time is artificial, it gets projected based on the start time + duration of
+ * the time span.
+ */
+public class TimeSpan implements Comparable<TimeSpan> {
```

Comment:
Wandering if we can do it internal.
Do we want to expose it to the user?

Reply:
Good point, let's not expose this - at least not in v0.